grails-app/migrations/remove_detection_matview.groovy

```diff
@@ -0,0 +1,101 @@
+databaseChangeLog = {
+    changeSet(author: "jburgess", id: "1420772532000-01") {
+
+        grailsChange {
+            change {
+                sql.execute("select drop_matview('detection_count_per_station_mv')")
+            }
+        }
+
+        dropView(viewName: 'detection_count_per_station')
+
+        dropView(viewName: 'public_detection_view')
+
+        grailsChange {
+            change {
+                sql.execute("select drop_matview('detection_extract_view_mv')")
+            }
+        }
+
+        dropView(viewName: 'detection_extract_view')
+
+        createView(
+            '''SELECT detection_view."timestamp",
+            detection_view.formatted_timestamp,
+            detection_view.station,
+            detection_view.station_id,
+            detection_view.location,
+            detection_view.latitude,
+            detection_view.longitude,
+            detection_view.receiver_name,
+            detection_view.sensor_id,
+            detection_view.species_name,
+            detection_view.uploader,
+            detection_view.transmitter_id,
+            detection_view.organisation,
+            detection_view.project,
+            detection_view.installation,
+            detection_view.spcode,
+            detection_view.animal_release_id,
+            detection_view.embargo_date,
+            detection_view.project_id,
+            detection_view.detection_id,
+            scramblepoint(detection_view.location) AS public_location,
+                CASE
+                    WHEN isreleaseembargoed(detection_view.animal_release_id) THEN 'embargoed'::text
+                    ELSE detection_view.species_name
+                END AS public_species_name,
+                CASE
+                    WHEN isreleaseembargoed(detection_view.animal_release_id) THEN 'embargoed'::character varying
+                    ELSE detection_view.spcode
+                END AS public_spcode
+            FROM detection_view;''',
+            viewName: 'public_detection_view'
+        )
+
+        // TODO: this view can most likely be optimised - it currently takes a long time (hours) to run.
```

Comment:
Note this TODO: I think this is worth investigating before deploying to production.  In theory, this view should only require a single scan of the detections table, which normally takes a few minutes or so.